A fresh checkout results in failing tests?

I wanted to look into what I think might be a date-related bug but checking out the repo and following the contributing steps results in failing tests. I'm not sure what I'm doing wrong, any help appreciated.

**Steps to Reproduce**

1. Checkout the repo
    ```sh
    $ git status
    On branch main
    Your branch is up to date with 'origin/main'.
    ```
2. Init submodule
    ```sh
    $ git submodule update --init
    $ git submodule status
     aa121de88de512680fe5752f0e9bfc07e311854c wasi/testdata (aa121de)
    ```
3. Run tests
    ```sh
    $ deno task test
    Task test deno test --doc --unstable --allow-all --parallel --coverage=./cov
    ./assert/assert_exists_test.ts => AssertExists ... ok (7ms)
    ./assert/assert_equals_test.ts => pass case ... ok (7ms)
    ...
    ./testing/snapshot_test.ts => SnapshotTest - createAssertSnapshot ... Composite ... ok (5ms)
    ./testing/snapshot_test.ts => SnapshotTest - createAssertSnapshot ... ok (26ms)
    
     ERRORS
    
    [std/datetime] dayOfYear => ./datetime/day_of_year_test.ts:189:6
    error: AssertionError: Values are not equal.
    
    
        [Diff] Actual / Expected
    
    
    -   90
    +   91
    
      throw new AssertionError(message);
            ^
        at assertEquals (file:///Users/gaggle/Projects/deno_std/assert/assert_equals.ts:53:9)
        at fn (file:///Users/gaggle/Projects/deno_std/datetime/day_of_year_test.ts:202:5)
    
     FAILURES
    
    [std/datetime] dayOfYear => ./datetime/day_of_year_test.ts:189:6
    
    FAILED | 1835 passed (2359 steps) | 1 failed | 11 ignored (24s)
    
    error: Test failed
    ```

**Expected behavior**

Tests to run to completion

**Environment**

<!-- please complete the following information -->

- OS: MacOS Ventura 13.5 (22G74), Intel Core i7
- deno version: 
    ```sh
    $ deno --version
    deno 1.36.2 (release, x86_64-apple-darwin)
    v8 11.7.439.1
    typescript 5.1.6
    ```

- std version: Does this refer to the sha of this repo I'm on?
    ```sh
    git rev-parse HEAD
    ee6856c2ed955e44cff6eb16c673f16642799fe6
    ```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -33,6 +33,13 @@ jobs:
       - name: Run tests canary
         run: deno task test
 
+      - name: Run timezone-dependent tests
+        run: |
+          TZ=Australia/Sydney deno test datetime
+          TZ=Europe/London deno test datetime
+          TZ=America/Toronto deno test datetime
+        if: matrix.os == 'ubuntu-22.04'
+
       - name: Type check browser compatible modules
         shell: bash
         run: deno task test:browser

--- a/datetime/day_of_year.ts
+++ b/datetime/day_of_year.ts
@@ -22,7 +22,8 @@ export function dayOfYear(date: Date): number {
   const yearStart = new Date(date);
 
   yearStart.setFullYear(date.getFullYear(), 0, 0);
-  const diff = date.getTime() - yearStart.getTime();
+  const diff = (date.getTime() - date.getTimezoneOffset() * 60 * 1000) -
+    (yearStart.getTime() - yearStart.getTimezoneOffset() * 60 * 1000);
 
   return Math.floor(diff / DAY);
 }

